src/hammer-vlsi/hammer_vlsi/hammer_tool.py

```diff
@@ -613,47 +697,58 @@ def make_resume_hook(step: str, location: HookLocation) -> HammerToolHookAction:
     def make_pre_pause_hook(step: str) -> HammerToolHookAction:
         """
         Create pause before the execution of the given step.
+        Note that only one pause hook may be present.
         """
-        return HammerTool.make_insertion_hook(step, HookLocation.InsertPreStep, HammerTool.make_pause_function())
+        return HammerTool.make_resume_pause_hook(step, HookLocation.PausePreStep)
 
     @staticmethod
     def make_post_pause_hook(step: str) -> HammerToolHookAction:
         """
         Create pause before the execution of the given step.
+        Note that only one pause hook may be present.
         """
-        return HammerTool.make_insertion_hook(step, HookLocation.InsertPostStep, HammerTool.make_pause_function())
+        return HammerTool.make_resume_pause_hook(step, HookLocation.PausePostStep)
 
     @staticmethod
     def make_pre_resume_hook(step: str) -> HammerToolHookAction:
         """
         Resume before the given step.
         Note that only one resume hook may be present.
         """
-        return HammerTool.make_resume_hook(step, HookLocation.ResumePreStep)
+        return HammerTool.make_resume_pause_hook(step, HookLocation.ResumePreStep)
 
     @staticmethod
     def make_post_resume_hook(step: str) -> HammerToolHookAction:
         """
         Resume after the given step.
         Note that only one resume hook may be present.
         """
-        return HammerTool.make_resume_hook(step, HookLocation.ResumePostStep)
+        return HammerTool.make_resume_pause_hook(step, HookLocation.ResumePostStep)
 
     @staticmethod
-    def make_from_to_hooks(from_step: Optional[str] = None,
-                           to_step: Optional[str] = None) -> List[HammerToolHookAction]:
-        """
-        Helper function to create a HammerToolHookAction list which will run from and to the given steps, inclusive.
-
-        :param from_step: Run from the given step, inclusive. Leave as None to resume from the beginning.
-        :param to_step: Run to the given step, inclusive. Leave as None to run to the end.
+    def make_start_stop_hooks(from_step: Optional[str] = None,
```

Comment:
While I agree that a generic form of the API is needed like this one, it would be good to have explicitly inclusive and exclusive variants e.g. `make_start_stop_hooks_inclusive` and `make_start_stop_hooks_exclusive`. This will help make it clearer as the generic API requires some double-taking to understand when reading.

Reply:
I disagree, because the directives are meant to be mixed, e.g. a `--from_step` (inclusive) can be paired with an `--until_step` (exclusive). Thus, it doesn't make sense to separate it out.

Also, this shouldn't be a user-facing API anyway, as it's abstracted under the CLIDriver.

I'll attempt to improve the documentation instead.

Reply:
So I understand the need for the usecase of mixing them; it's just that code like `make_start_stop_hooks("step5", None, "step5", None)` could be clearer if it were `make_start_stop_hooks_inclusive("step5",  "step5")` since then an API user won't have to guess where the "None"s should go. Alternatively if the signature was something like `make_start_stop_hooks(Optional[tuple of string, bool], Optional[tuple of string, bool])` where the bool indicates exclusiveness. This will make sure that we don't specify two froms or two tos.

Reply:
I took the `NamedTuple` approach to clear this up. Inclusivity takes precedence and `--only_step` takes precedence over all.